tests/integration.spec.ts

```diff
@@ -55,6 +55,20 @@ describe('jmap-client-ts', () => {
     await container.stop();
   });
 
+  it('should get error correctly', async () => {
+    let error = null;
+    try {
+      await client.mailbox_get({
+        accountId: 'unknown-account-id',
+        ids: null,
+      });
+    } catch (e) {
+      error = e;
```

Comment:
You don't have built in assertions to assert the thrown exception?

In java using the AssertJ lib I would writte:

```
assertThatThrownBy(() -> doX(y))
    .isInstanceOf(MyException.class);
```

Reply:
According to [documentation](https://jestjs.io/docs/expect#tothrowerror) and [this question on SO](https://stackoverflow.com/questions/48707111/asserting-against-thrown-error-objects-in-jest)
I cannot check the exception has the correct type.
I can check the exception is thrown, but the test would be less precise
Even if the form is not the best one, it is functional so I will keep it like this.

Reply:
So we are missing a decent assertion library ... :-/

Reply:
I reopened, I read a bit more of your stack overflow thread. I prefer this approach:

```
t('should throw', () => {

  const errorObj = {
    myError: {
      name: 'myError',
      desc: 'myDescription'
    }
  };

  const fn = () => {
    throw errorObj;
  }

  expect(() => fn()).toThrowError(new Error(errorObj));
});
```

Reply:
@chibenwa that would be a unit test, not integration test. I think first test should be unit test. Once it is tested, then one can move on to integration test.

Reply:
@chibenwa Error only takes the message (a string) as constructor parameter.